rewindable-index/src/RewindableIndex/Storable.hs

```diff
@@ -335,14 +335,7 @@ filterWithQueryInterval
   -> [StorableEvent h]
   -> [StorableEvent h]
 filterWithQueryInterval QEverything es = es
-filterWithQueryInterval (QInterval start end) es =
-  let es' = takeWhile (withPoint (\p -> p <= end)) es
-   in if not (null es') && withPoint (\p -> p >= start) (last es')
-      then es'
-      else []
-  where
-    withPoint :: (StorablePoint h -> Bool) -> StorableEvent h -> Bool
-    withPoint f e = let p = getPoint e in f p
+filterWithQueryInterval (QInterval start end) es = takeWhile (\e -> getPoint e <= end) $ dropWhile (\e -> getPoint e < start) es
```

Comment:
This would be a nice implementation, but it does not work properly. I think the comment may not be doing a very good job at explaining what this function does. Maybe I need to reword it.

Reply:
@eyeinsky That part needs to be reverted by I think

Reply:
Right, but the tests will fail then. The wider issue here is that it is unclear what an interval means and how could I test it.

Reply:
a928d3434, 4d2e1e617